Vamos a proceder a realizar un proyecto referido a Siniestro Viales. Para lo cual vamos a crear un dataset desde los archivos Excel

Procederemos a importar las librerias neceseria 

In [2]:
import pandas as pd

Vamos a crear un dataset df_homicidios a partir del archivo homicidios.xlsx

In [3]:
df_homicidios = pd.read_excel('../Data/homicidios.xlsx')

Vamos a ver el dataset

In [4]:
df_homicidios

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
691  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
692  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
693  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
694  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
695  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                           AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   
..                                         ...           ...  ...     ...   
691                          AV. RIESTRA Y MOM       AVENIDA  ...     NaN   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...     NaN   
693                        AV. GAONA Y TERRADA       AVENIDA  ...     NaN   
694                         AV. EVA PERON 4071       AVENIDA  ...  4071.0   
695                    PADRE CARLOS MUJICA 709         CALLE  ...   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
693   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
694    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
695  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    A

Voy a ver la información del dataset df_homicidios

In [5]:
df_homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Detectamos que el dataset contiene 696 filas y 21 columnas. Y luego de observar la informacion detectamos que no posee nulos 

Vamos a ver si el dataset contiene duplicados

In [6]:
duplicados = df_homicidios.duplicated()
print(duplicados)

0      False
1      False
2      False
3      False
4      False
       ...  
691    False
692    False
693    False
694    False
695    False
Length: 696, dtype: bool


No hay duplicados en el dataset df_homicidios

Eliminaré las columnas AAAA, MM y DD porque la información que contienen estas columnas está contenida en la columna FECHA

In [7]:
# Eliminar la columna 'AAAA'
df_homicidios = df_homicidios.drop('AAAA', axis=1)


In [8]:
# Eliminar la columna 'AAAA'
df_homicidios = df_homicidios.drop('MM', axis=1)

In [131]:
# Eliminar la columna 'AAAA'
df_homicidios = df_homicidios.drop('DD', axis=1)

In [9]:
df_homicidios

ID  N_VICTIMAS      FECHA  DD      HORA  HH  \
0    2016-0001           1 2016-01-01   1  04:00:00   4   
1    2016-0002           1 2016-01-02   2  01:15:00   1   
2    2016-0003           1 2016-01-03   3  07:00:00   7   
3    2016-0004           1 2016-01-10  10  00:00:00   0   
4    2016-0005           1 2016-01-21  21  05:20:00   5   
..         ...         ...        ...  ..       ...  ..   
691  2021-0093           1 2021-12-13  13  17:10:00  17   
692  2021-0094           1 2021-12-20  20  01:10:00   1   
693  2021-0095           1 2021-12-30  30  00:43:00   0   
694  2021-0096           1 2021-12-15  15  10:30:00  10   
695  2021-0097           1 2021-11-18  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ   
2                           AV ENTRE RIOS 2034       AVENIDA   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA   
..                                         ...           ...   
691                          AV. RIESTRA Y MOM       AVENIDA   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA   
693                        AV. GAONA Y TERRADA       AVENIDA   
694                         AV. EVA PERON 4071       AVENIDA   
695                    PADRE CARLOS MUJICA 709         CALLE   

                            Calle  Altura  \
0                PIEDRA BUENA AV.     NaN   
1                  PAZ, GRAL. AV.     NaN   
2                  ENTRE RIOS AV.  2034.0   
3                  LARRAZABAL AV.     NaN   
4                    SAN JUAN AV.     NaN   
..                            ...     ...   
691                   RIESTRA AV.     NaN   
692  DELLEPIANE, LUIS, TTE. GRAL.     NaN   
693                     GAONA AV.     NaN   
694                PERON, EVA AV.  4071.0   
695           PADRE CARLOS MUJICA   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada  COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.       9   
2                                  ENTRE RIOS AV. 2034       1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1   
..                                                 ...     ...   
691                                  RIESTRA AV. y MOM       7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.       9   
693                                GAONA AV. y TERRADA      11   
694                                PERON, EVA AV. 4071       9   
695                            PADRE CARLOS MUGICA 709       1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34

Voy a cambiar el tipo de dato de la Columna COMUNA de entero a objet. 

In [10]:
# Cambiar el tipo de la columna 'COMUNA' a 'object'
df_homicidios['COMUNA'] = df_homicidios['COMUNA'].astype('object')


In [11]:
df_homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   DD                     696 non-null    int64         
 4   HORA                   696 non-null    object        
 5   HH                     696 non-null    object        
 6   LUGAR_DEL_HECHO        696 non-null    object        
 7   TIPO_DE_CALLE          696 non-null    object        
 8   Calle                  695 non-null    object        
 9   Altura                 129 non-null    float64       
 10  Cruce                  525 non-null    object        
 11  Dirección Normalizada  688 non-null    object        
 12  COMUNA                 696 non-null    object        
 13  XY (C

Reemplazaré el tipo de calle en donde dice GRAL PAZ por AVENIDA

In [12]:
df_homicidios['TIPO_DE_CALLE']= df_homicidios['TIPO_DE_CALLE'].replace({'GRAL PAZ': 'AVENIDA'})

In [13]:
df_homicidios

ID  N_VICTIMAS      FECHA  DD      HORA  HH  \
0    2016-0001           1 2016-01-01   1  04:00:00   4   
1    2016-0002           1 2016-01-02   2  01:15:00   1   
2    2016-0003           1 2016-01-03   3  07:00:00   7   
3    2016-0004           1 2016-01-10  10  00:00:00   0   
4    2016-0005           1 2016-01-21  21  05:20:00   5   
..         ...         ...        ...  ..       ...  ..   
691  2021-0093           1 2021-12-13  13  17:10:00  17   
692  2021-0094           1 2021-12-20  20  01:10:00   1   
693  2021-0095           1 2021-12-30  30  00:43:00   0   
694  2021-0096           1 2021-12-15  15  10:30:00  10   
695  2021-0097           1 2021-11-18  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA   
1            AV GRAL PAZ  Y AV DE LOS CORRALES       AVENIDA   
2                           AV ENTRE RIOS 2034       AVENIDA   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA   
..                                         ...           ...   
691                          AV. RIESTRA Y MOM       AVENIDA   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA   
693                        AV. GAONA Y TERRADA       AVENIDA   
694                         AV. EVA PERON 4071       AVENIDA   
695                    PADRE CARLOS MUJICA 709         CALLE   

                            Calle  Altura  \
0                PIEDRA BUENA AV.     NaN   
1                  PAZ, GRAL. AV.     NaN   
2                  ENTRE RIOS AV.  2034.0   
3                  LARRAZABAL AV.     NaN   
4                    SAN JUAN AV.     NaN   
..                            ...     ...   
691                   RIESTRA AV.     NaN   
692  DELLEPIANE, LUIS, TTE. GRAL.     NaN   
693                     GAONA AV.     NaN   
694                PERON, EVA AV.  4071.0   
695           PADRE CARLOS MUJICA   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   

Vamos a crear un csv a partir del dataset df_homicilios

In [14]:
df_homicidios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   DD                     696 non-null    int64         
 4   HORA                   696 non-null    object        
 5   HH                     696 non-null    object        
 6   LUGAR_DEL_HECHO        696 non-null    object        
 7   TIPO_DE_CALLE          696 non-null    object        
 8   Calle                  695 non-null    object        
 9   Altura                 129 non-null    float64       
 10  Cruce                  525 non-null    object        
 11  Dirección Normalizada  688 non-null    object        
 12  COMUNA                 696 non-null    object        
 13  XY (C

In [15]:
df_homicidios.to_csv('homicidios.csv')

Creare un dataset df_lesiones a partir del archivo lesiones.xlsx

In [35]:
df_lesiones = pd.read_excel('../Data/lesiones.xlsx')

Procederé a ver el dataset 

In [36]:
df_lesiones

id  n_victimas  aaaa  mm  dd                fecha  \
0      LC-2019-0000179           1  2019   1   1  2019-01-01 00:00:00   
1      LC-2019-0000053           1  2019   1   1  2019-01-01 00:00:00   
2      LC-2019-0000063           1  2019   1   1  2019-01-01 00:00:00   
3      LC-2019-0000079           1  2019   1   1  2019-01-01 00:00:00   
4      LC-2019-0000082           4  2019   1   1  2019-01-01 00:00:00   
...                ...         ...   ...  ..  ..                  ...   
23780  LC-2021-0652849           1  2021  12  31  2021-12-31 00:00:00   
23781  LC-2021-0652865           2  2021  12  31  2021-12-31 00:00:00   
23782  LC-2021-0652907           1  2021  12  31  2021-12-31 00:00:00   
23783  LC-2021-0652921           1  2021  12  31  2021-12-31 00:00:00   
23784  LC-2020-0244615           1  2020   5  17            5/17/2020   

           hora franja_hora                 direccion_normalizada comuna  ...  \
0      09:00:00           9                                    SD     14  ...   
1      01:55:00           1                                    SD      8  ...   
2      02:00:00           2                                    SD      8  ...   
3      02:30:00           2                                    SD      7  ...   
4      04:30:00           4                                    SD      3  ...   
...         ...         ...                                   ...    ...  ...   
23780  19:30:00          19  ALBERDI, JUAN BAUTISTA AV. y GUAMINI      9  ...   
23781  19:40:00          19       ALBERDI, JUAN BAUTISTA AV. 4436      9  ...   
23782  20:00:00          20                                    SD      1  ...   
23783  22:00:00          22               LINIERS VIRREY y MORENO      5  ...   
23784  21:00:00          21                                    SD     SD  ...   

                latutid   victima             acusado  \
0            -34.559658  CICLISTA                  SD   
1            -34.669125      AUTO                  SD   
2            -34.677556        SD                  SD   
3            -34.647349    PEATON                  SD   
4            -34.604579      AUTO                  SD   
...                 ...       ...                 ...   
23780  -34.659713573880        SD                  SD   
23781  -34.641753304864        SD                  SD   
23782        -34.583083        SD                  SD   
23783  -34.614288229345      MOTO  TRANSPORTE PUBLICO   
23784       -3472890716      MOTO                AUTO   

                 participantes moto auto transporte_publico camion ciclista  \
0                  CICLISTA-SD   SD   SD                 SD     SD        x   
1                      AUTO-SD   SD    x                 SD     SD       SD   
2                        SD-SD   SD   SD                 SD     SD       SD   
3                    PEATON-SD    x   SD                 SD     SD       SD   
4                      AUTO-SD   SD   SD                  x     SD       SD   
...                        ...  ...  ...                ...    ...      ...   
23780                    SD-SD   SD   SD                 SD     SD       SD   
23781                    SD-SD   SD   SD                 SD     SD       SD   
23782                    SD-SD   SD   SD                 SD     SD       SD   
23783  MOTO-TRANSPORTE PUBLICO    x    0                  x      0        0   
23784                MOTO-AUTO    x    x                  0      0        0   

      gravedad  
0           SD  
1           SD  
2           SD  
3           SD  
4           SD  
...        ...  
23780       SD  
23781       SD  
23782       SD  
23783    GRAVE  
23784       SD  

[23785 rows x 27 columns]

Voy a ver la informacion de los datos del dataset df_lesiones

In [37]:
df_lesiones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23785 entries, 0 to 23784
Data columns (total 27 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     23785 non-null  object 
 1   n_victimas             23785 non-null  int64  
 2   aaaa                   23785 non-null  int64  
 3   mm                     23785 non-null  int64  
 4   dd                     23785 non-null  int64  
 5   fecha                  23785 non-null  object 
 6   hora                   23785 non-null  object 
 7   franja_hora            23780 non-null  object 
 8   direccion_normalizada  23732 non-null  object 
 9   comuna                 23616 non-null  object 
 10  tipo_calle             23785 non-null  object 
 11  otra_direccion         23785 non-null  object 
 12  calle                  12867 non-null  object 
 13  altura                 12771 non-null  float64
 14  cruce                  9407 non-null   object 
 15  ge

Vemos que el dataset no tiene nulos en su columna 'id'. Y que el dataset df_lesiones tiene 23785 filas y 27 columnas

Procederé a ver si posee duplicados teniendo en cuenta la columna id

In [38]:
duplicados = df_lesiones.duplicated(subset='id')
print(duplicados)

0        False
1        False
2        False
3        False
4        False
         ...  
23780    False
23781    False
23782    False
23783    False
23784    False
Length: 23785, dtype: bool


No hay duplicados en el dataset df_lesiones

Elimino las columnas 'aaaa'(año),'mm' (mes )y 'dd (día) porque este dato se encuentra en la columna fecha

In [39]:
# Eliminar la columna 'aaaa'
df_lesiones = df_lesiones.drop('aaaa', axis=1)

In [40]:
# Eliminar la columna 'mm'
df_lesiones = df_lesiones.drop('mm', axis=1)

In [41]:
# Eliminar la columna 'dd'
df_lesiones = df_lesiones.drop('dd', axis=1)

Cambiaré Gral Paz por Avenida

In [43]:
df_lesiones['tipo_calle']= df_lesiones['tipo_calle'].replace({'GRAL PAZ': 'AVENIDA'})

Vamos a crear un csv a partir del dataset df_lesiones

In [44]:
df_lesiones.to_csv('lesiones.csv')